api/app/Models/PoolCandidate.php

```diff
@@ -323,4 +324,39 @@ public function scopeExpiryFilter(Builder $query, ?array $args) {
         }
         return $query;
     }
+
+   /* accessor to obtain pool candidate status, additional logic exists to override database field sometimes*/
+   public function getPoolCandidateStatusAttribute()
+   {
+        // pull info
+        $submittedAt = $this->submitted_at;
+        $expiryDate = $this->expiry_date;
+        $currentTime = date("Y-m-d H:i:s");
+        $isExpired = $currentTime > $expiryDate ? true : false;
+
+        // to avoid recursively calling the accessor, it is important to SELECT it outside of the regular way via direct DB
+        // grab self Id, run a DB statement and then grab the value out of the resulting object in an array
+        $candidateId = $this->id;
+        $results = DB::select('select pool_candidate_status from pool_candidates where id = :id', ['id' => $candidateId]);
+        $candidateStatus = $results[0]->pool_candidate_status;
```

Comment:
Just for my own knowledge, can you explain why using eloquent causes problems here?

Reply:
this was my guess from troubleshooting, attempting to access the property seemed to keep erroring while the other properties were accessible
so I thought it was a case of calling the accessor in the accessor causing weirdness and recursion

Reply:
In example, setting it through Eloquent like so

![image](https://user-images.githubusercontent.com/40485260/186224490-84c95f4a-acc9-4b21-a95c-e5094684887b.png)

results in this error

![image](https://user-images.githubusercontent.com/40485260/186224205-6979efa8-41e2-4c3f-8b2c-b6e3a13650cd.png)

despite all those properties being present here

![image](https://user-images.githubusercontent.com/40485260/186224417-b2102dd8-6162-4166-9b1a-f84d29283635.png)

Reply:
stuff like that I was a bit shaky on, so that's why I was specifically requesting a look-over by multiple people

Reply:
So, that is likely because you are defining the accessor so you can't access it within its own accessor. Laravel helps with this by passing the original value in as the argument for the accessor.

### Example

REF: https://laravel.com/docs/8.x/eloquent-mutators#accessors-and-mutators

```php
public function getPoolCandidateStatusAttribute($value)
{
  // $value === $this->pool_candidate_status
}
```

Reply:
thank you! that was exactly what I needed!
done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/3744/commits/09cbc88e7a3090c934393fdb2315215412f4834a